apps/src/code-studio/hiddenStageRedux.js

```diff
@@ -2,48 +2,107 @@
  * Reducer and actions for stage lock info. This includes the teacher panel on
  * the course overview page, and the stage locking dialog.
  */
-
+import $ from 'jquery';
 import { INIT_PROGRESS } from './progressRedux';
 import experiments from '@cdo/apps/experiments';
 
-export const TOGGLE_HIDDEN = 'hiddenStage/TOGGLE_HIDDEN';
+export const UPDATE_HIDDEN_STAGES = 'hiddenStage/UPDATE_HIDDEN_STAGES';
+export const ALLOW_HIDEABLE = 'hiddenStage/ALLOW_HIDEABLE';
 
 export const hiddenStagesEnabled = () => experiments.isEnabled('hiddenStages');
 
+// State is a key indicating whether or not we've received hidden stage info
+// from the server, and then a set of stageIds mapped to bools that are true
+// if that stage is hidden
+const initialState = {
+  initialized: false
```

Comment:
I don't love this state construction, where it ends up looking something like:
```
state = {
   initialized: true,
   123: true,
   456: false
};
```
Better would probably be to add another level of nesting:
```
state = {
   initialized: true,
   stages: {
     123: true,
     456: false
  }
};
```
but this makes updates a lot more awkward.

Thoughts? I might be able to make the updates less awkward by using something like immutable.js